sky/clouds/vsphere.py

```diff
@@ -278,19 +279,23 @@ def _check_compute_credentials(
             cls) -> Tuple[bool, Optional[Union[str, Dict[str, str]]]]:
         """Checks if the user has access credentials to
         vSphere's compute service."""
-
+        dependency_error_msg = (
+            'vSphere dependencies are not installed. '
+            'Run the following commands:'
+            f'\n{cls._INDENT_PREFIX}  $ pip install skypilot[vSphere]'
+            f'\n{cls._INDENT_PREFIX}Credentials may also need to be set. '
+            'For more details. See https://docs.skypilot.co/en/latest/getting-started/installation.html#vmware-vsphere'  # pylint: disable=line-too-long
+        )
         try:
             # pylint: disable=import-outside-toplevel,unused-import
             # Check pyVmomi installation.
-            import pyVmomi
+            pyvmomi_spec = importlib_util.find_spec('pyVmomi')
+            if pyvmomi_spec is None:
+                return False, dependency_error_msg
         except (ImportError, subprocess.CalledProcessError) as e:
             return False, (
-                'vSphere dependencies are not installed. '
-                'Run the following commands:'
-                f'\n{cls._INDENT_PREFIX}  $ pip install skypilot[vSphere]'
-                f'\n{cls._INDENT_PREFIX}Credentials may also need to be set. '
-                'For more details. See https://docs.skypilot.co/en/latest/getting-started/installation.html#vmware-vsphere'  # pylint: disable=line-too-long
-                f'{common_utils.format_exception(e, use_bracket=True)}')
+                dependency_error_msg +
+                f' {common_utils.format_exception(e, use_bracket=True)}')
 
```

Comment:
Why would this reduce memory? It seems the later code will still call into the `get_vsphere_credentials`?